src/main/kotlin/org/rust/lang/core/completion/RsAttributeCompletionProvider.kt

```diff
@@ -70,11 +65,12 @@ object RsAttributeCompletionProvider : RsCompletionProvider() {
         onStructLike to "non_exhaustive"
     ).flatMap { entry -> entry.value.split(' ').map { attrName -> RustAttribute(attrName, entry.key) } }
 
-    override fun addCompletions(parameters: CompletionParameters,
-                                context: ProcessingContext,
-                                result: CompletionResultSet) {
-
-        val elem = parameters.position.superParent(RsPsiPattern.META_ITEM_IDENTIFIER_DEPTH)
+    override fun addCompletions(
+        parameters: CompletionParameters,
+        context: ProcessingContext,
+        result: CompletionResultSet
+    ) {
+        val elem = context[META_ITEM_ATTR]?.parent ?: return
 
         val suggestions = attributes
             .filter { it.appliesTo.accepts(parameters.position) && elem.attrMetaItems.none { item -> item == it.name } }
```

Comment:
In fact, we can replace `it.appliesTo.accepts(parameters.position)` with `it.appliesTo.accepts(elem)` to reduce number of computation here because all our patterns in `attributes` list provide some restriction only for items but it:
- potentially restricts us from writing patterns that take into account attributes characteristics
- requires to refactor all our `on` patterns and the corresponding tests
so Im't not sure if it's worth it